disdrodb/tests/test_check_metadata.py

```diff
@@ -1,14 +1,297 @@
 import os
 import pytest
 import yaml
+import random
 
 
-from disdrodb.l0 import check_metadata
+from disdrodb.l0 import check_metadata, metadata
 
 
 PATH_TEST_FOLDERS_FILES = os.path.join(os.path.dirname(os.path.realpath(__file__)), "pytest_files")
 
 
+def test_check_metadata_geolocation():
+    # Test missing longitude and latitude
+    with pytest.raises(ValueError):
+        metadata = {"platform_type": "fixed"}
+        check_metadata.check_metadata_geolocation(metadata)
+
+    # Test non-numeric longitude
+    with pytest.raises(TypeError):
+        metadata = {"longitude": "not_a_number", "latitude": 20, "platform_type": "fixed"}
+        check_metadata.check_metadata_geolocation(metadata)
+
+    # Test non-numeric latitude
+    with pytest.raises(TypeError):
+        metadata = {"longitude": 10, "latitude": "not_a_number", "platform_type": "fixed"}
+        check_metadata.check_metadata_geolocation(metadata)
+
+    # Test mobile platform with wrong coordinates
+    with pytest.raises(ValueError):
+        metadata = {"longitude": 10, "latitude": 20, "platform_type": "mobile"}
+        check_metadata.check_metadata_geolocation(metadata)
+
+    # Test fixed platform with missing latitude
+    with pytest.raises(ValueError):
+        metadata = {"longitude": 10, "latitude": -9999, "platform_type": "fixed"}
+        check_metadata.check_metadata_geolocation(metadata)
+
+    # Test fixed platform with missing longitude
+    with pytest.raises(ValueError):
+        metadata = {"longitude": -9999, "latitude": 20, "platform_type": "fixed"}
+        check_metadata.check_metadata_geolocation(metadata)
+
+    # Test invalid longitude value
+    with pytest.raises(ValueError):
+        metadata = {"longitude": 200, "latitude": 20, "platform_type": "fixed"}
+        check_metadata.check_metadata_geolocation(metadata)
+
+    # Test invalid latitude value
+    with pytest.raises(ValueError):
+        metadata = {"longitude": 10, "latitude": -100, "platform_type": "fixed"}
+        check_metadata.check_metadata_geolocation(metadata)
+
+    # Test valid metadata
+    metadata = {"longitude": 10, "latitude": 20, "platform_type": "fixed"}
+    assert check_metadata.check_metadata_geolocation(metadata) is None
+
+
+def create_fake_metadata_file(
+    tmp_path, yaml_file_name, yaml_dict, data_source="data_source", campaign_name="campaign_name"

```

Comment:
`tmp_path` directory is necessary or can be created/defined automatically inside the function ?

Reply:
It seems to be the best practice (it is unique to each test function invocation). https://docs.pytest.org/en/7.3.x/reference/reference.html#tmp-path